metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -355,6 +360,60 @@ def get_datasource_urn_from_id(
             )
         raise ValueError("Could not construct dataset URN")
 
+    def parse_owner_payload(self, payload: Dict[str, Any]) -> Dict[str, str]:
+        owners_dict = {}
+        for owner_data in payload["result"]:
+            email = owner_data["extra"]["email"]
+            value = owner_data["value"]
+
+            if value and email:
+                owners_dict[value] = email
+        return owners_dict
+
+    def build_preset_owner_dict(self) -> Dict[str, str]:
+        owners_dict = {}
+        dataset_payload = self.get_all_entity_owners("dataset")
+        chart_payload = self.get_all_entity_owners("chart")
+        dashboard_payload = self.get_all_entity_owners("dashboard")
+
+        owners_dict.update(self.parse_owner_payload(dataset_payload))
+        owners_dict.update(self.parse_owner_payload(chart_payload))
+        owners_dict.update(self.parse_owner_payload(dashboard_payload))
+        return owners_dict
+
+    def build_owners_urn_list(self, data: Dict[str, Any]) -> List[str]:
+        owners_urn_list = []
+        for owner in data.get("owners", []):
+            owner_id = owner.get("id")
+            owner_email = self.owners_dict.get(owner_id)
+            if owner_email is not None:
+                owners_urn = make_user_urn(owner_email)
+                owners_urn_list.append(owners_urn)
+        return owners_urn_list
+
+    def get_all_entity_owners(self, entity: str) -> Dict[str, Any]:
+        current_page = 1
+        total_owners = PAGE_SIZE
+        all_owners = []
+
+        while (current_page - 1) * PAGE_SIZE <= total_owners:
+            full_owners_response = self.session.get(
+                f"{self.config.connect_uri}/api/v1/{entity}/related/owners",
+                params=f"q=(page:{current_page},page_size:{PAGE_SIZE})",
+            )
+            if full_owners_response.status_code != 200:
+                logger.warning(
```

Comment:
This logs a warning and then raises the exception, which will fail the ingestion, right?

IMO we can either log a warning and proceed without owners, ensuring the ingestion doesn't fail, or report it as an error and halt the ingestion process.